.github/workflows/step_pre-commit.yml

```diff
@@ -0,0 +1,16 @@
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit:
+    name: Run pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - uses: pre-commit/action@v3.0.0
```

Comment:
This could be a good opportunity to try using https://pre-commit.ci/, to which tmt could also later also potentially switch. wdyt?

Reply:
Just be aware what caching is used in `pre-commit/action` vs `pre-commit`. This is relevant for `mypy`, but that should be resolved by moving the testing infrastructure to `tmt` anyway. The other difference is on the `auto-update` and if you want each individual service to notify of dependency changes, or if you want to consolidate them with something like `renovate.json`. Here's a starter file I use consolidating `pre-commit` and GH-Actions updates:
```json
{
	"$schema": "https://docs.renovatebot.com/renovate-schema.json",
	"packageRules": [
		{
			"groupName": "CI and devDependencies",
			"matchManagers": ["github-actions", "pre-commit"]
		}
	],
	"separateMajorMinor": false,
	"extends": [
		"config:recommended",
		"schedule:weekly",
		":enablePreCommit",
		":semanticCommitTypeAll(chore)"
	]
}
```

Reply:
Thanks. Glad you have experience with the action. I have close to none :)  
Would be good to put together a list of pros and cons to see if it make sense considering.